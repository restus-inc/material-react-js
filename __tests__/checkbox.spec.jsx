/* global jest, describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../lib/checkbox';

jest.mock('@material/checkbox');
jest.mock('@material/form-field');

describe('Checkbox component', () => {
  it('supports checkbox', () => {
    let component = renderer.create(
      <Checkbox/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox className="baz" disabled={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox id="foo" label="bar" checked={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox label="bar" className="baz" indeterminate={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox id="foo" label="bar" className="baz" disabled={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('supports touch target', () => {
    let component = renderer.create(
      <Checkbox supportsTouch={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox className="baz" disabled={true} supportsTouch={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox id="foo" label="bar" checked={true} supportsTouch={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox label="bar" className="baz" indeterminate={true} supportsTouch={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(
      <Checkbox id="foo" label="bar" className="baz" disabled={true} supportsTouch={true}/>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
