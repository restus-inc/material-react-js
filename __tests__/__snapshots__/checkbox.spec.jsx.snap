// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component supports checkbox 1`] = `
<div
  className="mdc-checkbox"
>
  <input
    className="mdc-checkbox__native-control"
    type="checkbox"
  />
  <div
    className="mdc-checkbox__background"
  >
    <svg
      className="mdc-checkbox__checkmark"
      viewBox="0 0 24 24"
    >
      <path
        className="mdc-checkbox__checkmark-path"
        d="M1.73,12.91 8.1,19.28 22.79,4.59"
        fill="none"
      />
    </svg>
    <div
      className="mdc-checkbox__mixedmark"
    />
  </div>
  <div
    className="mdc-checkbox__ripple"
  />
</div>
`;

exports[`Checkbox component supports checkbox 2`] = `
<div
  className="mdc-checkbox mdc-checkbox--disabled baz"
>
  <input
    className="mdc-checkbox__native-control"
    disabled={true}
    type="checkbox"
  />
  <div
    className="mdc-checkbox__background"
  >
    <svg
      className="mdc-checkbox__checkmark"
      viewBox="0 0 24 24"
    >
      <path
        className="mdc-checkbox__checkmark-path"
        d="M1.73,12.91 8.1,19.28 22.79,4.59"
        fill="none"
      />
    </svg>
    <div
      className="mdc-checkbox__mixedmark"
    />
  </div>
  <div
    className="mdc-checkbox__ripple"
  />
</div>
`;

exports[`Checkbox component supports checkbox 3`] = `
<div
  className="mdc-form-field"
>
  <div
    className="mdc-checkbox"
  >
    <input
      checked={true}
      className="mdc-checkbox__native-control"
      id="foo"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
    <div
      className="mdc-checkbox__ripple"
    />
  </div>
  <label
    htmlFor="foo"
  >
    bar
  </label>
</div>
`;

exports[`Checkbox component supports checkbox 4`] = `
<div
  className="mdc-form-field baz"
>
  <div
    className="mdc-checkbox"
  >
    <input
      className="mdc-checkbox__native-control"
      data-indeterminate="true"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
    <div
      className="mdc-checkbox__ripple"
    />
  </div>
  <label>
    bar
  </label>
</div>
`;

exports[`Checkbox component supports checkbox 5`] = `
<div
  className="mdc-form-field baz"
>
  <div
    className="mdc-checkbox mdc-checkbox--disabled"
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={true}
      id="foo"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
    <div
      className="mdc-checkbox__ripple"
    />
  </div>
  <label
    htmlFor="foo"
  >
    bar
  </label>
</div>
`;

exports[`Checkbox component supports touch target 1`] = `
<div
  className="mdc-touch-target-wrapper"
>
  <div
    className="mdc-checkbox mdc-checkbox--touch"
  >
    <input
      className="mdc-checkbox__native-control"
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
    <div
      className="mdc-checkbox__ripple"
    />
  </div>
</div>
`;

exports[`Checkbox component supports touch target 2`] = `
<div
  className="mdc-touch-target-wrapper baz"
>
  <div
    className="mdc-checkbox mdc-checkbox--disabled mdc-checkbox--touch"
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={true}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
    <div
      className="mdc-checkbox__ripple"
    />
  </div>
</div>
`;

exports[`Checkbox component supports touch target 3`] = `
<div
  className="mdc-touch-target-wrapper"
>
  <div
    className="mdc-form-field"
  >
    <div
      className="mdc-checkbox mdc-checkbox--touch"
    >
      <input
        checked={true}
        className="mdc-checkbox__native-control"
        id="foo"
        type="checkbox"
      />
      <div
        className="mdc-checkbox__background"
      >
        <svg
          className="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            className="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
        <div
          className="mdc-checkbox__mixedmark"
        />
      </div>
      <div
        className="mdc-checkbox__ripple"
      />
    </div>
    <label
      htmlFor="foo"
    >
      bar
    </label>
  </div>
</div>
`;

exports[`Checkbox component supports touch target 4`] = `
<div
  className="mdc-touch-target-wrapper baz"
>
  <div
    className="mdc-form-field"
  >
    <div
      className="mdc-checkbox mdc-checkbox--touch"
    >
      <input
        className="mdc-checkbox__native-control"
        data-indeterminate="true"
        type="checkbox"
      />
      <div
        className="mdc-checkbox__background"
      >
        <svg
          className="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            className="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
        <div
          className="mdc-checkbox__mixedmark"
        />
      </div>
      <div
        className="mdc-checkbox__ripple"
      />
    </div>
    <label>
      bar
    </label>
  </div>
</div>
`;

exports[`Checkbox component supports touch target 5`] = `
<div
  className="mdc-touch-target-wrapper baz"
>
  <div
    className="mdc-form-field"
  >
    <div
      className="mdc-checkbox mdc-checkbox--disabled mdc-checkbox--touch"
    >
      <input
        className="mdc-checkbox__native-control"
        disabled={true}
        id="foo"
        type="checkbox"
      />
      <div
        className="mdc-checkbox__background"
      >
        <svg
          className="mdc-checkbox__checkmark"
          viewBox="0 0 24 24"
        >
          <path
            className="mdc-checkbox__checkmark-path"
            d="M1.73,12.91 8.1,19.28 22.79,4.59"
            fill="none"
          />
        </svg>
        <div
          className="mdc-checkbox__mixedmark"
        />
      </div>
      <div
        className="mdc-checkbox__ripple"
      />
    </div>
    <label
      htmlFor="foo"
    >
      bar
    </label>
  </div>
</div>
`;
